File: easy/trick_or_treat.pl
use strict;
use integer;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    next if ( $line =~ m/^\s$/ );
    my ( $v, $z, $w, $h ) = ( $line =~
          /Vampires: (\d+), Zombies: (\d+), Witches: (\d+), Houses: (\d+)/ );
    my $r =
      ( $v + $z + $w == 0 )
      ? 0
      : ( $v * 3 + $z * 4 + $w * 5 ) * $h / ( $v + $z + $w );
    print "$r\n";
}
close(INFILE);

File: easy/trick_or_treat.rb
p = /Vampires: (\d+), Zombies: (\d+), Witches: (\d+), Houses: (\d+)/
File.open(ARGV[0]).each_line do |line|
  s = line.scan(p)[0].map(&:to_i)
  if s[0] + s[1] + s[2] == 0
    puts 0
  else
    puts s[3] * (s[0] * 3 + s[1] * 4 + s[2] * 5) / (s[0] + s[1] + s[2])
  end
end
